### PR TITLE
I8 create navbar component

### LINKED ISSUE #8: Create Navbar Component
## Basic Information
This component must be responsive.
Must include
- WADL Logo
- Login/Logout button

### PR DESCRIPTION
**Pull request by LemonTheSour**
## Change Summary
-Implemented a basic NavBar component with a Home and Login button. Both buttons just link to the index page at present. 
-Added a logos file to assets with several of the logos in various colours.
-Updated default.vue in layout to display NavBar across all pages by default. 

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
closes #8 

# Related issue

- Resolve #8

### CODE REVIEW

File: components/NavBar/NavBar.vue [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <span class="bg-gold shadow flex position:relative place-items-center">
```
Comment (ChantelleKerr) [inline components/NavBar/NavBar.vue:2]: span should be changed to <nav>

File: components/NavBar/LoginButton.vue [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+<script lang="ts">
+export default {
+  props: {
+    buttonText: {
+      type: String,
+      default: () => "LoginButton",
+    },
+  },
+};
```
Comment (ChantelleKerr) [inline components/NavBar/LoginButton.vue:15]: You should use the composition API - Add setup to the script element and remove export default
https://vuejs.org/guide/extras/composition-api-faq.html#what-is-composition-api

You can handle props by using the defineprops function
https://vuejs.org/api/sfc-script-setup.html#defineprops-defineemits

File: components/NavBar/NavBar.vue [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  components: {
+    HomeButton,
+    LoginButton,
+  },
+  methods: {
+    onClick() {
+      console.log("login");
+    },
+  },
+};
```
Comment (ChantelleKerr) [inline components/NavBar/NavBar.vue:22]: Please change this to using the composition API. 
The onClick function can be removed as it's not used.

File: components/NavBar/HomeButton.vue [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<template>
+  <NuxtLink to="/">
+    <img class="h-16 py-1" src="~/assets/logos/TextlessTransparentLogo.png" />
+  </NuxtLink>
+</template>
+
+<script lang="ts">
+export default {};
+</script>
```
Comment (ChantelleKerr) [inline components/NavBar/HomeButton.vue:9]: This can be removed

File: pages/index.vue [hunk 5]
```diff
@@ -1,5 +1,9 @@
...
-    <NuxtWelcome />
-  </div>
+  <div></div>
 </template>
+
+<script lang="ts">
+export default {
+  name: "Index",
+};
+</script>
```
Comment (ChantelleKerr) [inline pages/index.vue:9]: remove the export default and change script to use setup :)

**Review by ChantelleKerr — CHANGES_REQUESTED** [review]
Hi Michael, great job so far! I love that you broke up the navbar into different components and also used a correct folder structure. 
I have requested some small changes and I recommend looking up the composition API (I've provided some links)
Committee has also decided that we will not be using typescript so replace lang="ts" with setup.

It looks great btw!

File: components/NavBar/LoginButton.vue [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<template>
+  <NuxtLink to="/">
+    <button class="text-3xl">{{ buttonText }}</button>
```
Comment (ChantelleKerr) [inline components/NavBar/LoginButton.vue:3]: It looks better with text-2xl

File: components/NavBar/NavBar.vue [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<template>
+  <nav class="bg-gold shadow flex position:relative place-items-center">
+    <HomeButton class="pl-3" />
+    <LoginButton class="absolute right-3" button-text="Login" />
```
Comment (ChantelleKerr) [inline components/NavBar/NavBar.vue:4]: Change button-text to buttonText. We need to stay consistent by using camelCase :)

**Review by ChantelleKerr — CHANGES_REQUESTED** [review]
Thanks for doing the changes. I missed a few minor things in the previous PR review. Once those are fixed I can merge into main :D

File: components/NavBar/NavBar.vue (hunk 7 above)
Comment (ChantelleKerr) [reply components/NavBar/NavBar.vue:4]: Did this get done? @LemonTheSour

Comment (LemonTheSour) [reply components/NavBar/NavBar.vue:4]: Just pushed up now

**Review by ChantelleKerr — APPROVED** [review]
Good job! It looks amazing :D

**Pull request merged**